ethexe/contracts/foundry.toml

```diff
@@ -8,8 +8,13 @@ ffi = true
 ast = true
 build_info = true
 extra_output = ["storageLayout"]
-# Warning (3628): This contract has a payable fallback function, but no receive ether function.
-ignored_warnings_from = ["src/MirrorProxy.sol"]
+# Some warnings should be omitted
+ignored_warnings_from = [
+    # Warning (3628): This contract has a payable fallback function, but no receive ether function
+    "src/MirrorProxy.sol",
+]
+# Enable new EVM codegen
+via_ir = true
```

Comment:
Compilation time has increased by 3x, but overall it will save us gas when deploying contracts to mainnet. Only the ERC20 token was regressed (which is unlikely to be updated as often as router).
<details>
  <summary>Comparison table</summary>

```
|-----------------------------|----------|----------|
| New codegen enabled?        | via_ir=0 | via_ir=1 |
| Compilation Time            | 5.52s    | 15.89s   |
| Contract                    | Size (B) | Size (B) |
|-----------------------------|----------|----------|
| Address                     |       85 |       57 |
| BeaconProxy                 |      283 |      291 |
| Clones                      |       85 |       57 |
| Core                        |       85 |       57 |
| Defender                    |       85 |       57 |
| DefenderDeploy              |       85 |       57 |
| ECDSA                       |       85 |       57 |
| ERC1967Proxy                |      163 |      130 |
| ERC1967Utils                |       85 |       57 |
| Errors                      |       85 |       57 |
| Math                        |       85 |       57 |
| MessageHashUtils            |       85 |       57 |
| Mirror                      |    3,998 |    3,119 |
| MirrorProxy                 |      399 |      396 |
| Panic                       |       85 |       57 |
| ProxyAdmin                  |    1,018 |      881 |
| Router                      |   11,917 |   10,389 |
| SafeCast                    |       85 |       57 |
| SignedMath                  |       85 |       57 |
| StorageSlot                 |       85 |       57 |
| Strings                     |       85 |       57 |
| TransparentUpgradeableProxy |    1,115 |      793 |
| UnsafeUpgrades              |       85 |       57 |
| UpgradeableBeacon           |      644 |      605 |
| Upgrades                    |       85 |       57 |
| Utils                       |       85 |       57 |
| Versions                    |       85 |       57 |
| WrappedVara                 |    6,116 |    7,360 |
| strings                     |       85 |       57 |
```

</details>